backend/ranking.js

```diff
@@ -0,0 +1,158 @@
+import { Stock } from "./models/stock.js";
+import { Follow } from "./models/follow.js";
+import { Like } from "./models/like.js";
+import { Dislike } from "./models/dislike.js";
+import { compareStocksByPoints } from "./utils.js";
+
+
+const FOLLOW_POINTS = 25;
+const LIKE_POINTS = 20;
+const DISLIKE_POINTS = -20;
+
+
+export async function RankingOptimized(user) {
+    try {
+        const [stocksInDatabase, follows, likes, dislikes] = await Promise.all([
+            Stock.findAll(),
+            Follow.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            }),
+            Like.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            }),
+            Dislike.findAll({
+                where: { UserId: user.id },
+                include: [{ model: Stock }]
+            })
+        ]);
+
+        if (stocksInDatabase.length < 10) {
+            return {
+                status: 422,
+                error: "You need at least 10 stocks in the database to access ranking."
+            };
+        }
+
+        const followsMap = {};
+
+        follows.forEach((follow) => {
+            followsMap[follow.Stock.ticker] = true;
+        });
+
+        const likesMap = {};
+
+        likes.forEach((like) => {
+            likesMap[like.Stock.ticker] = true;
+        });
+
+        const dislikesMap = {};
+
+        dislikes.forEach((dislike) => {
+            dislikesMap[dislike.Stock.ticker] = true;
+        });
+
```

Comment:
Writing this with forEach works fine here and looks readable as well. 

More like food for thought: In JS, there's always multiple interesting ways to implement something. There's a neat function called `reduce` (read more [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce)) that could have better performance than `forEach`. We could express the same result doing something like this:
```
dislikesMap = dislikes.reduce((result, current) => {
    result[current.Stock.ticker] = true;
    return result;
}, {}); 
```

Reply:
Adding to that:

Sometimes I call the returning variable `accum` or `accumulator` because it accumulates the final result.
```
dislikesMap = dislikes.reduce((accum, current) => { 
  accum[current.Stock.ticker] = true; 
  return accum; 
}, {});
```
But `result` also works :)

About the readability tradeoff, it is a subjective topic, off course, but once you get really used to it and the rest of the declarative, functional programming like idioms, then it becomes easier to read. Same as you probably think using map makes the code easier to read than using for loop.